src/clj/rems/locales.clj

```diff
@@ -1,42 +1,77 @@
 (ns rems.locales
   {:ns-tracker/resource-deps ["translations/en.edn" "translations/fi.edn"]}
   (:require [clojure.java.io :as io]
+            [clojure.set :as set]
             [clojure.string :as str]
+            [clojure.tools.logging :as log]
             [mount.core :refer [defstate]]
-            [rems.common-util :refer [deep-merge]]
+            [rems.common-util :refer [deep-merge recursive-keys]]
             [rems.config :refer [env]])
   (:import (java.io FileNotFoundException)))
 
-(defn- translations-from-file [filename dir]
-  (let [file (when dir
-               (io/file dir filename))
-        resource-path (str dir filename)
-        resource (io/resource resource-path)
-        file-contents (cond
-                        (and file (.exists file)) file
-                        resource resource
-                        :else (throw (FileNotFoundException.
-                                      (if file
-                                        (str "translations could not be found in " file " file or " resource-path " resource")
-                                        (str "translations could not be found in " resource-path " resource and " :translations-directory " was not set")))))]
-    (read-string (slurp file-contents))))
+(defn- file-or-resource
+  "Given a path (a String or File), resolve it to either an existing file or existing resource.
+  Returns a File or URL object, or nil if file nor resource can be found.
+  Prefers files over resources."
+  [path]
+  (let [file (io/file path)
+        resource (io/resource (str file))]
+    (cond
+      (and file (.exists file)) file
+      resource resource
+      :else nil)))
+
+(defn- translations-from-file [file]
+  (if-let [chosen (file-or-resource file)]
+    (read-string (slurp chosen))
+    (throw (FileNotFoundException.
+            (str "translations could not be found in file or resource \"" file "\"")))))
 
 (defn- extra-translations-path [theme-path]
-  (let [path (str/join "/" (butlast (str/split theme-path #"/"))) ;;Theme-path is of form /foo/bar/theme.edn
-        translations-path (str path "/extra-translations/")]
-    translations-path))
+  (-> theme-path
+      (io/file)
+      (.getParentFile)
+      (io/file "extra-translations")))
```

Comment:
I think the following would communicate the meaning better, because the thing being passed through `->` changes type (theme file vs. parent directory vs. extra translations directory). See https://tonsky.me/blog/readable-clojure/#dont-spare-names 
```clj
(let [parent-dir (.getParentFile (io/file theme-path))]
  (io/file parent-dir "extra-translations"))

Reply:
fair enough